lib/mlir/Pipes/MLIRLLVMHelpers.h

```diff
@@ -0,0 +1,64 @@
+#pragma once
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include "mlir/Dialect/LLVMIR/LLVMDialect.h"
+
+namespace revng::mlir_llvm {
+
+using LocType = mlir::FusedLocWith<mlir::LLVM::DISubprogramAttr>;
+
+inline mlir::Block &getModuleBlock(mlir::ModuleOp Module) {
+  revng_assert(Module);
+  revng_assert(Module->getNumRegions() == 1);
+  return Module->getRegion(0).front();
+}
+
+inline decltype(auto) getModuleOperations(mlir::ModuleOp Module) {
+  return getModuleBlock(Module).getOperations();
+}
+
+inline std::optional<llvm::StringRef> getLinkageName(mlir::Operation *const O) {
+  if (const auto L = mlir::dyn_cast<LocType>(O->getLoc()))
+    return L.getMetadata().getLinkageName();
+  return std::nullopt;
+}
+
+inline std::optional<llvm::StringRef>
+getFunctionName(mlir::FunctionOpInterface F) {
+  if (const auto Name = getLinkageName(F)) {
```

Comment:
AFAIU we can have "custom" attributes on operations, at this point there's no point in abusing linkage name.
If you're not up for doing it now, at least leave a big TODO.

Reply:
I got rid of any uses of the linkage name and added the `revng.function.entry` `StringAttr` attribute containing the `MetaAddress`.